### PR TITLE
Migrate n

### PR DESCRIPTION
**Pull request by tiagoantao**
* [X] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [X] This PR adds a new recipe.
* [X] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

This PR is a new recipe for migrate-n. Can someone @bioconda/core have a look please?

### CODE REVIEW

File: recipes/migrate-n/run_test.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/python
+
+import os
+import sys
+
+if os.system('migrate-n -h') != 65280:
```
Comment (bgruening) [inline recipes/migrate-n/run_test.py]: can you migrate this into the yaml file and do something like this `migrate-n -h | grep "Foo"` ... with Foo a text included in the help message. This way the error code is taken from the grep step and it should be 0.

File: recipes/migrate-n/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package:
+  name: migrate-n
+  version: "3.6.11"
```
Comment (bgruening) [inline recipes/migrate-n/meta.yaml]: you could use jinja templates for the version string

File: recipes/migrate-n/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+build:
+  number: 0
+  skip: True  # [not linux]
+
+source:
+  fn: migrate-3.6.11.src.tar.gz
+  url: http://popgen.sc.fsu.edu/currentversions/migrate-3.6.11.src.tar.gz
+  md5: 4e85552db1e73dc394042176e710508c
```
Comment (bgruening) [inline recipes/migrate-n/meta.yaml]: sha256 is possible.

File: recipes/migrate-n/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    - zlib {{ CONDA_ZLIB }}*
+
+about:
+  home: http://popgen.sc.fsu.edu/Migrate/Migrate-n.html
+  license: MIT
+  summary: Population Genetics - Panmixia and Migration detection
+
+extra:
+  identifiers:
+    - doi:10.1093/bioinformatics/btu033
```
Comment (bgruening) [inline recipes/migrate-n/meta.yaml]: Yeah! Thanks for adding the DOI!

File: recipes/migrate-n/build.sh [hunk 5]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+
+cd src
+
+./configure
+sed -i 's/#include <xlocale.h>//' data.c
```
Comment (bgruening) [inline recipes/migrate-n/build.sh]: `-i.bak` if you want to be compatible with OSX

**Comment by tiagoantao** [comment]
Thanks @bgruening , as usual you are amazing. Could you have another look to see if it looks OK?

**Comment by tiagoantao** [comment]
Thanks for the help @bgruening

**Pull request merged**